File: twoSum/twoSum.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
	public int[] twoSum(int[] nums, int target) {
		int[] result = new int[2];
		Map<Integer, Integer> map = new HashMap<Integer, Integer>();
		
		for(int i = 0; i < nums.length; i++) {
			if(map.get(target - nums[i]) != null) {
				result[0] = map.get(target - nums[i]) + 1;
				result[1] = i + 1;
				break;
			} else {
				map.put(nums[i], i);
			}
		}
		return result;
	}
}

File: twoSum/twoSum.cpp
class Solution {
	public:
		vector<int> twoSum(vector<int> &nums, int target) {
			vector<int> result;
			map<int, int> hash;
			for(int i = 0; i < nums.size(); i++) {
				if(hash.find(target - nums[i]) != hash.end()) {
					result.push_back(hash[target - nums[i]] + 1);
					result.push_back(i + 1);
					break;
				} else {
					hash[nums[i]] = i;
				}
			}			
			return result;
		}
};
